docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,407 @@
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.connection.duration` | Histogram | `s` | The duration of successfully established outbound HTTP connections. [1] |
+
+**[1]:** Meter name is `System.Net.Http`.
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.client.connection.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [1] | `1.1`; `2` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to. [2] | `example.com` | Required |
+| [`server.port`](../general/attributes.md) | int | Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to. [3] | `80`; `8080`; `443` | Conditionally Required: [4] |
+| [`server.socket.address`](../general/attributes.md) | string | Server address of the socket connection - IP address or Unix domain socket name. [5] | `10.5.3.2` | Recommended: If different than `server.address`. |
+| [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https`; `ftp` | Recommended |
+
+**[1]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+**[2]:** Determined by using the first of the following that applies
+
+- Host identifier of the [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource)
+  if it's sent in absolute-form
+- Host identifier of the `Host` header
+
+**[3]:** When [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) is absolute URI, `server.port` MUST match URI port identifier, otherwise it MUST match `Host` header port identifier.
+
+**[4]:** If not default (`80` for `http` scheme, `443` for `https`).
+
+**[5]:** When observed from the client side, this SHOULD represent the immediate server peer address.
+When observed from the server side, this SHOULD represent the physical server address.
+<!-- endsemconv -->
+
+### Metric: `http.client.request.duration`
+
+Client request duration measures time it takes for the call time to underlying client handler to complete request.
+If request was sent with `HttpCompletionOption.ResponseContentRead` option (the default value) then metric records time to last byte, otherwise (if `HttpCompletionOption.ResponseHeadersRead` was set), metric records time it took to receive response headers.
```

Comment:
this is not correct - client duration measures time to headers (per https://github.com/open-telemetry/semantic-conventions/pull/205#discussion_r1319194521)

Reply:
There are two contradicting goals:

If we wanted the span to cover content buffering, we would need to implement metrics in `HttpClient` / `HttpMessageInvoker` instead of a `MetricsHandler`, since handlers are not aware of `HttpCompletionOption`.

If we implemented metrics in `HttpClient`/`HttpMessageInvoker` it wouldn't be possible to log separate spans for redirections, as we agreed in https://github.com/open-telemetry/opentelemetry-specification/discussions/3462.